src/goto-programs/builtin_functions.cpp

```diff
@@ -580,7 +580,7 @@ void goto_convertt::do_function_call_symbol(
   }
   else if(base_name == "__ESBMC_assert")
   {
-    if(arguments.size() != 2)
+    if(arguments.size() > 2)
```

Comment:
Could you add some comments here? Why the size of the arguments must be greater than 2?

Reply:
What happens if we get something different?

Reply:
I've changed the messages and comments for:

1 - Default assertion
2 - Assertion with a specific message
>2 - An error

The behaviour will be: `__ESBMC_assert(cond) ---> __ESBMC_assert(cond, "ESBMC assertion")`

Reply:
I suggest reverting it since it doesn't add any meaningful information. It can be simply an assert